src/utils/debounce.ts

```diff
@@ -0,0 +1,34 @@
+import { FArguments } from '../types';
+
+const defaultDebounceMap = new WeakMap<
+  Function,
+  ReturnType<typeof setTimeout>
+>();
+
+/**
+ * Create a debounced copy of a function
+ *
+ * @param fn Function to debounce
+ * @param timeout how long (in ms) to debounce
+ */
+export function debounce<F extends (...args: any[]) => void>(
+  fn: F,
+  timeout: number,
+  state: WeakMap<Function, ReturnType<typeof setTimeout>> = defaultDebounceMap,
+): (...args: FArguments<F>) => Promise<void> {
+  return async function debounced(...args: FArguments<F>): Promise<void> {
```

Comment:
What is actually being awaited on here?

Reply:
great point. I could just leave this as a non-async promise-returning function.

Reply:
@mike-north what does the promise returned fulfill or reject with?

Reply:
I was originally playing with the idea of debounce handling return values by returning a promise. There's no immediate need for this, so I've removed the return value entirely. Fix is in #24